api/client/profile.go

```diff
@@ -212,11 +213,16 @@ func ProfileFromDir(dir string, name string) (*Profile, error) {
 		}
 	}
 
-	return ProfileFromFile(filepath.Join(dir, name+".yaml"))
+	cp, err := profileFromFile(filepath.Join(dir, name+".yaml"))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cp.Dir = dir
```

Comment:
Do you only need to define that when using `ProfileFromDir`? profileFromFile can also set it and it will be consistent:
```go
var cp *Profile
if err := yaml.Unmarshal(bytes, &cp); err != nil {
	return nil, trace.Wrap(err)
}
cp.Dir = filepath.Dir(filePath)
```